lib/rucio/daemons/replicarecoverer/suspicious_replica_recoverer.py

```diff
@@ -80,7 +82,54 @@ def check_suspicious_policy(policy: dict[str, str], file_metadata_datatype: str,
     return action
 
 
-def declare_suspicious_replicas_bad(once: bool = False, younger_than: int = 5, nattempts: int = 5, vos: Optional[list[str]] = None, limit_suspicious_files_on_rse: int = 5, json_file_name: str = "/opt/rucio/etc/suspicious_replica_recoverer.json", sleep_time: int = 3600, active_mode: bool = False) -> None:
+def parse_replica_datatype(
+        did_name_expression: Optional[str],
+        file_name: str,
+        rse_key: str,
+        logger: Callable) -> Optional[str]:
```

Comment:
```suggestion
        logger: LoggerFunction) -> Optional[str]:
```

With LoggerFunction being imported from `rucio.common.types`